core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -278,16 +280,18 @@ void toSerializableWithExtraRead(DistributedStorage storage)
           continue;
         }
         // Check if read records are not changed
-        if (!currentReadSetByScan.contains(readSet.get(key).get())) {
+        if (!currentReadSet.contains(readSet.get(key).get())) {
           throwExceptionDueToAntiDependency();
         }
-        validatedReadSetByScan.add(key);
+        validatedReadSet.add(key);
       }
 
       // Check if the size of a read set by scan is not changed
-      if (currentReadSetByScan.size() != validatedReadSetByScan.size()) {
+      if (currentReadSet.size() != validatedReadSet.size()) {
         throwExceptionDueToAntiDependency();
       }
+
+      validatedReadSetByScan.addAll(validatedReadSet);
```

Comment:
And we need a validated read set that has all the results of all the Scans for the validation for Gets.